Allow in-source builds, we can just nuke, and reextract if
builds need to be repeated. (also it's a pain to do out of
source builds)
---
diff -u a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2016-08-01 18:25:06.753718317 -0400
+++ b/CMakeLists.txt	2016-08-01 18:25:29.061706962 -0400
@@ -123,20 +123,7 @@
   message(FATAL_ERROR "Building shared libraries on Windows needs MinGW")
 endif ()
 
-# Disallow in-source builds
-if ("${PROJECT_BINARY_DIR}" STREQUAL "${PROJECT_SOURCE_DIR}")
-  message(FATAL_ERROR
-    "
-In-source builds are not permitted!  Make a separate folder for"
-    " building, e.g.,"
-    "
-  mkdir build; cd build; cmake .."
-    "
-Before that, remove the files created by this failed run with"
-    "
-  rm -rf CMakeCache.txt CMakeFiles")
-endif ()
-
+# Allow in-source builds
 
 # Add local cmake modules
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${PROJECT_SOURCE_DIR}/builds/cmake)
